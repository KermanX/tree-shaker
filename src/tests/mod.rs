use crate::{TreeShakeConfig, TreeShakeOptions};
use insta::{assert_snapshot, glob};
use oxc::{codegen::CodegenOptions, minifier::MinifierOptions};
use std::fs;

fn tree_shake(input: String) -> String {
  let do_minify = input.contains("@minify");
  let react_jsx = input.contains("@react-jsx");
  let result = crate::tree_shake(
    input,
    TreeShakeOptions {
      config: TreeShakeConfig::default().with_react_jsx(react_jsx),
      minify_options: do_minify.then(|| MinifierOptions::default()),
      codegen_options: CodegenOptions::default(),
    },
  );
  result.codegen_return.code
}

#[test]
fn test() {
  glob!("fixtures/**/*.js", |path| {
    let input = fs::read_to_string(path).unwrap();
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
      assert_snapshot!(tree_shake(input));
    })
  });
}
