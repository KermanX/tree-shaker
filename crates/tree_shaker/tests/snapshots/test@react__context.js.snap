---
source: src/tests/mod.rs
expression: tree_shake(input)
input_file: src/tests/fixtures/react/context.js
---
import React from "react";
export function case_provided() {
	const MyContext = React.createContext();
	function Inner() {
		return <div>{"world"}</div>;
	}
	return function main() {
		return <MyContext.Provider>
        <MyContext.Provider>
          <Inner />
        </MyContext.Provider>
      </MyContext.Provider>;
	};
}
export function case_not_provided() {
	const MyContext1 = React.createContext();
	function Inner() {
		return <div>{"default-2"}</div>;
	}
	return function main() {
		return <MyContext1.Provider>
        <Inner />
      </MyContext1.Provider>;
	};
}
export function case_consumed() {
	const MyContext = React.createContext("default");
	function Inner() {
		const value = React.useContext(MyContext);
		return <div>{value}</div>;
	}
	lostTrack(Inner);
	return function main() {
		return <MyContext.Provider value="hello">
        <UnknownComponent />
      </MyContext.Provider>;
	};
}
